core/src/main/scala/kafka/admin/ReassignPartitionsCommand.scala

```diff
@@ -575,7 +596,11 @@ class ReassignPartitionsCommand(zkUtils: ZkUtils,
   def reassignPartitions(throttle: Throttle = NoThrottle, timeoutMs: Long = 10000L): Boolean = {
     maybeThrottle(throttle)
     try {
-      val validPartitions = proposedPartitionAssignment.filter { case (p, _) => validatePartition(zkUtils, p.topic, p.partition) }
+      val topics = proposedPartitionAssignment.map(_._1.topic).toSeq.distinct
```

Comment:
There are a few problems with this:

- The method validatePartition() is no longer used. We should either put this new logic into Skipping reassignment of partition(), or remove validatePartition()
- This is change of behavior because we no longer have the log statement in the validatePartition() for partitions that does not exist in the zookeeper.
- It will be simpler and more intuitive to make existingTopicAndPartitions a set of partitions instead a map. Then validPartitions will something like `proposedPartitionAssignment.filter(existingTopicAndPartitions)`